src/osx/carbon/font.cpp

```diff
@@ -909,42 +1041,87 @@ wxString wxNativeFontInfo::ToString() const
     wxString s;
 
     s.Printf(wxT("%d;%d;%d;%d;%d;%d;%d;%s;%d"),
-             1,                                 // version
-             m_pointSize,
-             m_family,
-             (int)m_style,
-             (int)m_weight,
-             m_underlined,
-             m_strikethrough,
-             m_faceName.GetData(),
-             (int)m_encoding);
+        1, // version
+        GetPointSize(),
+        GetFamily(),
+        (int)GetStyle(),
+        (int)GetWeight(),
+        GetUnderlined(),
+        GetStrikethrough(),
+        GetFaceName().GetData(),
+        (int)GetEncoding());
 
     return s;
 }
 
 int wxNativeFontInfo::GetPointSize() const
 {
-    return m_pointSize;
+    return wxRound(GetFractionalPointSize());
+}
+
+float wxNativeFontInfo::GetFractionalPointSize() const
+{
+    return m_ctSize;
 }
 
 wxFontStyle wxNativeFontInfo::GetStyle() const
 {
     return m_style;
 }
 
+int wxNativeFontInfo::GetNumericWeight() const
+{
+    return CTWeightToWX(m_ctWeight /* GetCTWeight(m_descriptor)*/);
+}
+
 wxFontWeight wxNativeFontInfo::GetWeight() const
 {
-    return m_weight;
+    // round to nearest hundredth = wxFONTWEIGHT_ constant
+    int weight = ((GetNumericWeight() + 50) / 100) * 100;
+
+    if (weight < wxFONTWEIGHT_THIN)
+        weight = wxFONTWEIGHT_THIN;
+    if (weight > wxFONTWEIGHT_MAX)
+        weight = wxFONTWEIGHT_MAX;
+
+    return (wxFontWeight)weight;
 }
 
 bool wxNativeFontInfo::GetUnderlined() const
 {
     return m_underlined;
 }
 
+wxString wxNativeFontInfo::GetFamilyName() const
+{
+    return m_familyName;
+}
+
+wxString wxNativeFontInfo::GetStyleName() const
+{
+    return m_styleName;
+}
+
 wxString wxNativeFontInfo::GetFaceName() const
 {
-    return m_faceName;
+#if 0
```

Comment:
Could you use `wxLogTrace("ctfont", ...)` instead? This could stay enabled in all builds unless this function is very performance-critical, but I don't think it is (and if it is, let's put this inside `#if wxDEBUG_LEVEL >= 2` which is still better than `#if 0`).

Reply:
committed